rq/worker.py

```diff
@@ -160,11 +160,21 @@ def __init__(self, queues, name=None,
 
     def validate_queues(self):
         """Sanity check for the given queues."""
-        if not iterable(self.queues):
-            raise ValueError('Argument queues not iterable.')
         for queue in self.queues:
             if not isinstance(queue, self.queue_class):
-                raise NoQueueError('Give each worker at least one Queue.')
+                raise NoQueueError('{} is not a queue'.format(queue))
+
+    def process_queue_args(self, queue_args):
+        """ allow for a variety arguements to queues """
+        if isinstance(queue_args, text_type):
+            return self.queue_class(name = queue_args)
+        else:
+            queue_args = [queue_args] if isinstance(queue_args, text_type) else queue_args
```

Comment:
I think we still need it. queue_args could be a list and there is still a chance for the user to pass in something that is not a queue.

As the code is currently written it will flatten iterables that are nested one deep. (maybe this is unnecessary?)

```Python
Worker([['foo']]) #this will work
Worker([[['foo']]]) #this will not work
```
I would be happy to write more tests if that would clear things up.